rbac/management/group/view.py

```diff
@@ -681,7 +690,9 @@ def principals(self, request: Request, uuid: Optional[UUID] = None):
                 else:
                     principals.append(specified_principal)
 
-            self.protect_system_groups("add principals", group)
+            with transaction.atomic():
+                group = self.get_object()
+                self.protect_system_groups("add principals", group)
 
             if not request.user.admin:
```

Comment:
TODO: Move everything under here under the same atomic block

Reply:
Done 👍